### PR TITLE
feat(integrations): add support for Circle So

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe your changes

- Add support for circle

## Issue ticket number and link
[EXT-114](https://linear.app/nango/issue/EXT-114/add-support-for-circle)

### CODE REVIEW

File: packages/shared/providers.yaml [hunk 1]
```diff
@@ -492,6 +492,15 @@ chorus:
     proxy:
         base_url: https://chorus.ai
     docs: https://docs.nango.dev/integrations/all/chorus
+circle:
+    categories:
+        - communication
+    auth_mode: API_KEY
+    proxy:
```
Comment (hassan254-prog) [inline packages/shared/providers.yaml:499]: `proxy.verification` won't work here as the provider always returns `status: 200`. I had to manually test with a small sync script.

File: docs-v2/integrations/all/circle.mdx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+title: Circle
```
Comment (bodinsamuel) [inline docs-v2/integrations/all/circle.mdx]: can we rename that to Circle.so? I thought we were talking about circle ci

File: docs-v2/integrations/all/circle.mdx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+sidebarTitle: Circle
+---
+
+API configurations: [`circle`](https://nango.dev/providers.yaml)
```
Comment (bodinsamuel) [inline docs-v2/integrations/all/circle.mdx]: something like this
```suggestion
API configurations: [`circle-so`](https://nango.dev/providers.yaml)
```

**Review by bodinsamuel — APPROVED** [review]
lgtm except naming :D

File: docs-v2/integrations/all/circle.mdx (hunk 3 above)
Comment (hassan254-prog) [reply docs-v2/integrations/all/circle.mdx]: @bodinsamuel ,that makes perfect sense for distinguishing between the two. I have updated this in the latest [commit](https://github.com/NangoHQ/nango/pull/2577/commits/45e36a2368a9bf464331fc72f4c41e1d59884531).

File: packages/shared/providers.yaml (hunk 1 above)
Comment (khaliqgant) [reply packages/shared/providers.yaml:499]: What does it return if it fails? An error in the payload? Anything in the headers?

File: docs-v2/integrations/all/circle.mdx (hunk 3 above)
Comment (bodinsamuel) [reply docs-v2/integrations/all/circle.mdx]: thanks 🙏🏻

**Review by bodinsamuel — APPROVED** [review]

**Review by TBonnin — APPROVED** [review]

File: packages/shared/providers.yaml (hunk 1 above)
Comment (hassan254-prog) [reply packages/shared/providers.yaml:499]: @khaliqgant , the response returns with a status of `200` even after sending the wrong APIkey. A good example is: `{ "status": "unauthorized", "message": "Your account could not be authenticated." }`

**Pull request merged**